numba/cuda/cuda_paths.py

```diff
@@ -25,8 +24,6 @@ def _find_valid_path(options):
 
 def _get_libdevice_path_decision():
     options = [
-        ('NUMBAPRO_LIBDEVICE', get_numbapro_envvar('NUMBAPRO_LIBDEVICE')),
```

Comment:
Maybe it would be helpful for the user to keep the `get_numbapro_envvar(...)` and make it raise when the env-var is detected.

Reply:
I would prefer a loud error message, but not actually raise an exception.  I think a lot of people using the NUMBAPRO env vars in the past were trying to use their system CUDA, which we now detect anyway.

Reply:
How about a warning and the message point at the docs? Would that suffice?

Reply:
Yes, that seems reasonable.  We won't actually use the env var for anything, but we'll keep pointing people to the right place.  (There a lot of old docs out there to overcome.)

Reply:
Added in 4ab78b3. `NUMBAPRO_*` envvars will now cause a `NumbaWarning` to be displayed and have no effect on library loading.